theopensystemslab/planx-new
# feat: create a content migration endpoint to add new `PropertyInformation` component to flows in correct position

depends on #1401 for testing, sorry it makes for such a large diff! please just focus on **api.planx.uk/** changes here

i've noted some known outstanding to-do's below, but would appreciate a preliminary code review here to sense-check the direction this is going! 

todo: 
- make this a batch operation, rather than single flow
- more meaningful response & error handling - eg if flow doesn't have FindProperty (silently skips for now), etc
- figure out why API tests are passing locally, but failing on CI and linking my query mock to the BOPS test file path??

testing: 
- create or open an existing flow that has a FindProperty component, publish it
- run GET `/admin/migrate/splitFindProperty/:flowId` in a browser tab (must be signed into editor)
- reload your flow in the editor
  - confirm there is now a PropertyInformation node displaying directly after FindProperty
  - click "Check for changes to publish" - it should not pick up any changes
  - add/move/remove a node from the graph, confirm you can keep editing the flow without error (this will confirm it's okay to skip adding an associated operation for the migration) 
- edge casey stuff: make sure this works regardless of _where_ FindProperty is positioned in the graph (eg for new prior approval services it's on a branch); make sure it works with and without a published version, etc
